### PR TITLE
ref: merge requirements-pre-commit into dev

### PR DESCRIPTION
**Pull request by joshuarli**
This merges pc reqs into dev to simplify some things ahead of introducing lockfiles. It's slightly more inefficient, but those inefficiences aren't really significant and will only shrink over time as prebuilt wheels and cached virtualenvs will be introduced in the future.

The ideal would be to have pre-commit manage everything, but we'd need something like `pre-commit exec` and then we could alias like, `black` to `pre-commit exec black`. @asottile-sentry lemme know if you'd accept a PR like that to upstream, although I understand it goes against pre-commit's design (pre-commit doesn't want to be something like pipx).

### CODE REVIEW

**Review by armenzg — APPROVED** [review]
This makes the setup simpler 👍🏻

**Comment by asottile-sentry** [comment]
@joshuarli `pre-commit run black --files $filename` is ~close to that already -- but yeah it's not meant to be an arbitrary execution engine

I had another thought on how we could still have the separate versions but have pre-commit manage the actual environments by syncing the versions to a separate requirements file that people can optionally install

or we could just make a vs-code extension which runs pre-commit and that'd probably make most people happy (?)

**Review by asottile-sentry — APPROVED** [review]

File: scripts/lib.sh [hunk 1]
```diff
@@ -145,7 +145,10 @@ setup-git() {
         echo 'Please run `make setup-pyenv` to install the required Python 3 version.'
         exit 1
     )
-    pip install -r requirements-pre-commit.txt
```
Comment (asottile-sentry) [inline scripts/lib.sh:148]: might be simpler to do `-r requirements-dev.txt` here -- would need fewer changes elsewhere

Comment (joshuarli) [reply scripts/lib.sh:148]: Eh, I kind of wanna roll it all up into `make install-py-dev` once we make that fast.

Comment (joshuarli) [reply scripts/lib.sh:148]: Actulaly nvm, need to do this to fix backend lint.

File: .github/workflows/backend.yml [hunk 2]
```diff
@@ -154,8 +154,6 @@ jobs:
         uses: ./.github/actions/setup-python
         with:
           python-version: ${{ matrix.python-version }}
-          # Note this uses a different cache key than other backend-based workflows because this workflows dependencies are different
-          cache-files-hash: ${{ hashFiles('requirements-pre-commit.txt') }}
```
Comment (asottile-sentry) [inline .github/workflows/backend.yml:158]: this'll need to be back to `requirements-dev.txt` I think

Comment (joshuarli) [reply .github/workflows/backend.yml:158]:
```
  cache-files-hash:
    description: 'A single hash for a set of files. Used for caching.'
    required: false
    default: ${{ hashFiles('requirements-*.txt') }}
```

Comment (joshuarli) [reply .github/workflows/backend.yml:158]: Oh wait, this is backend lint.

**Review by asottile-sentry — APPROVED** [review]

**Pull request merged**